### PR TITLE
add docs for google map field

### PR DESCRIPTION
**Pull request by pxpm**

### CODE REVIEW

**Comment by tabacitu** [comment]
Pedro let's please update the docs, so that "saving in two inputs" teaches devs to use Field Events, rather than hidden fields.

**Comment by jorgetwgroup** [comment]
Look ready documentation.

Cheers

File: 5.x/crud-fields.md [hunk 1]
```diff
@@ -466,6 +466,73 @@ Input preview:
 
 <hr>
 
+<a name="google-map"></a>
+### google_map
```
Comment (tabacitu) [inline 5.x/crud-fields.md:470]: I don't think this should be under FREE fields 👀 It's a PRO field, isn't it?

File: 5.x/crud-fields.md [hunk 2]
```diff
@@ -466,6 +466,73 @@ Input preview:
...
+                'lng' => $location->lng,
+                'full_address' => $location->formatted_address ?? ''
+            ];
+        }
+    );
+}
+
+```
+
+Input preview:
```
Comment (tabacitu) [inline 5.x/crud-fields.md:530]: Let's upload an image for it here, inside the Github conversation. 
Then we can take that URL and introduce it in the markdown 😉

File: 5.x/crud-fields.md [hunk 3]
```diff
@@ -466,6 +466,73 @@ Input preview:
...
+// 
+protected function location(): \Illuminate\Database\Eloquent\Casts\Attribute
+{
+    return \Illuminate\Database\Eloquent\Casts\Attribute::make(
+        get: fn ($value, $attributes) => json_encode([
+            'lat' => $attributes['lat'], 
+            'lng' => $attributes['lng'], 
+            'formatted_address' => $attributes['full_address'] ?? ''
+            ],
+             JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR),
```
Comment (tabacitu) [inline 5.x/crud-fields.md:516]: This looks weird to me on a separate line
```suggestion
            ], JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR),
```

File: 5.x/crud-fields.md [hunk 4]
```diff
@@ -466,6 +466,73 @@ Input preview:
...
+        set: function($value) {
+            $location = json_decode($value);
+            return [
+                'lat' => $location->lat,
+                'lng' => $location->lng,
+                'full_address' => $location->formatted_address ?? ''
+            ];
+        }
```
Comment (tabacitu) [inline 5.x/crud-fields.md:524]: If the getter is using arrow functions, shouldn't the setter too? Let's choose one and stick with it.

File: 5.x/crud-fields.md [hunk 5]
```diff
@@ -466,6 +466,73 @@ Input preview:
 
 <hr>
 
+<a name="google-map"></a>
+### google_map
+
+Uses the Google Places API to generate a Map where user can navigate and select a position to store `Latitude` (**lat**), `Longitude` (**lng**) and the address string (**formatted_address**). By default it saves a string in dabase with `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`.
```
Comment (tabacitu) [inline 5.x/crud-fields.md:472]: Sorry but no. This is NOT what the field does. Why are we writing from a technical perspective here? We should write from a USER perspective.

```suggestion
Shows a map and allows the user to navigate and select a position on that map (using the Google Places API). The field stores the latitude, longitude and the address string as a JSON in the database ( eg. `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`). If you want to save the info in separate db columns, continue reading below.
```

File: 5.x/crud-fields.md [hunk 6]
```diff
@@ -466,6 +466,73 @@ Input preview:
 
 <hr>
 
+<a name="google-map"></a>
+### google_map
+
+Uses the Google Places API to generate a Map where user can navigate and select a position to store `Latitude` (**lat**), `Longitude` (**lng**) and the address string (**formatted_address**). By default it saves a string in dabase with `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`. 
+**Note**: If you want two save in separate fields continue reading below.
```
Comment (tabacitu) [inline 5.x/crud-fields.md:473]:
```suggestion
```

**Review by tabacitu — CHANGES_REQUESTED** [review]

**Comment by tabacitu** [comment]
Can I just say I absolutely LOVE the fact that @pxpm chose Lisbon for the screenshot location? Just like Jorge chose Chile for the `phone` field 😀 I love that these fields have a nod to its creator 😉 👏👏👏

**Pull request merged**